spec/lib/submission_runners/haskell_spec.rb

```diff
@@ -0,0 +1,133 @@
+    let(:submission) {
+      Submission.create_from_file({
+        problem:  problem,
+        account:  account,
+        filename: fixtures.join("#{problem_name}.hs"),
+      }).tap(&:validate!)
+    }
+
+    subject(:runner) { described_class.new submission }
+
+    context 'good entry submission' do
+      let(:problem_name) { 'compiles_and_runs' }
+
+      it 'builds' do
+        b = runner.build
+        expect(b.err).to eq('')
+        expect(b.out).to_not eq(problem.output)
+        expect(b.success?).to be_truthy
+      end
+
+      it 'runs' do
+        runner.build
+        r = runner.run
+        expect(r.err).to eq('')
+        expect(r.out).to eq(problem.output)
+        expect(r.success?).to be_truthy
+      end
+
+      it 'builds and runs with call' do
+        runner.call
+        expect(runner.output).to eq(problem.output)
+        expect(runner.output_type).to eq('success')
+        expect(runner.run_succeeded).to be_truthy
+      end
+    end
+
+    context 'failing entry submission' do
+      let(:problem_name) { 'compiles_and_runs_wrong_answer' }
+
+      it 'builds' do
+        b = runner.build
+        expect(b.err).to eq('')
+        expect(b.out).to_not eq(problem.output)
+        expect(b.success?).to be_truthy
+      end
+
+      it 'runs' do
+        runner.build
+        r = runner.run
+        expect(r.err).to eq('')
+        expect(r.out).to_not eq(problem.output)
+        expect(r.success?).to be_truthy
+      end
+
+      it 'builds and runs with call' do
+        runner.call
+        expect(runner.output).to_not eq(problem.output)
+        expect(runner.output_type).to eq('success')
+        expect(runner.run_succeeded).to be_truthy
+      end
+    end
+
+    context 'compiles and doesn\'t run entry submission' do
+      let(:problem_name) { 'compiles_and_doesnt_run' }
+
+      it 'builds' do
+        b = runner.build
+        expect(b.err).to eq('')
+        expect(b.out).to_not eq(problem.output)
+        expect(b.success?).to be_truthy
+      end
+
+      it 'runs' do
+        runner.build
+        r = runner.run
+        # expect(r.err).to_not eq('')
+        expect(r.out).to_not eq(problem.output)
+        expect(r.success?).to be_falsey
+      end
+
+      it 'builds and runs with call' do
+        runner.call
+        expect(runner.output).to_not eq(problem.output)
+        expect(runner.output_type).to eq('run_failure')
+        expect(runner.run_succeeded).to be_falsey
+      end
+    end
+
+    context 'doesn\'t compile entry submission' do
+      let(:problem_name) { 'doesnt_compile' }
+
+      it 'builds' do
+        b = runner.build
+        expect(b.err).to_not eq('')
+        expect(b.out).to_not eq(problem.output)
+        expect(b.success?).to be_falsey
+      end
+
+      it 'runs' do
+        # Reports a successful run here even though compilation failed.
+        runner.build
+        r = runner.run
+        expect(r.out).to_not eq(problem.output)
+        expect(r.success?).to be_falsey
```

Comment:
This is the line that's having the issue.

Reply:
Since the compilation need to and does fail for this scenario, perhaps we don't need to test or care about running results afterwards.  That doesn't make sense for our system since we should be just stopping at the first failure point anyways.